### PR TITLE
[NPU] [Bugfix] Wan quantization fix

### LINKED ISSUE #24518: [Bug] [NPU] Wan2.2-T2V-A14B-Diffusers-w8a8 does not work
### Checklist

- [ ] I searched related issues but found no solution.
- [ ] The bug persists in the latest version.
- [ ] Issues without environment info and a minimal reproducible demo are hard to resolve and may receive no feedback.
- [ ] If this is not a bug report but a general question, please start a discussion at https://github.com/sgl-project/sglang/discussions. Otherwise, it will be closed.
- [ ] Please use English. Otherwise, it will be closed.

### Describe the bug

After merging https://github.com/sgl-project/sglang/pull/23625 and changing prefixes 

Image

now the quantization scheme cannot be recognized correctly (since the prefixes that are being searched are missing from the quant config), because of this, an error occurs:

Image

### Reproduction

SGLANG_CACHE_DIT_FN=2 SGLANG_CACHE_DIT_BN=1 SGLANG_CACHE_DIT_WARMUP=4 SGLANG_CACHE_DIT_RDT=0.4 SGLANG_CACHE_DIT_MC=4 SGLANG_CACHE_DIT_TAYLORSEER=true SGLANG_CACHE_DIT_TS_ORDER=2 SGLANG_CACHE_DIT_ENABLED=true  sglang generate --model-path ./weights/Wan2.2-T2V-A14B-Diffusers-w8a8/ --prompt "Two anthropomorphic cats in comfy boxing gear and bright gloves fight intensely on a spotlighted stage." --height 720 --width 1280 --tp-size 2 --sp-degree 2 --num-gpus 4 --num-frames 81 --num-inference-steps 40 --warmup true

### Environment

Python: 3.11.15 (main, Mar 11 2026, 17:11:19) [GCC 14.3.0]
NPU available: True
NPU 0,1,2,3,4,5,6,7,8,9,10,11,12,13,14,15: Ascend910_9362
CANN_HOME: /usr/local/Ascend/cann-8.5.0
CANN: Not Available
BiSheng: 2025-12-30T10:27:21+08:00 clang version 15.0.5 (clang-5c68a1cb1231 flang-5c68a1cb1231)
Ascend Driver Version: 25.5.1
PyTorch: 2.10.0+cpu
sglang: 0.5.12.dev101+gc8bc23522
sglang-kernel: Module Not Found
flashinfer_python: Module Not Found
flashinfer_cubin: Module Not Found
flashinfer_jit_cache: Module Not Found
triton: Module Not Found
transformers: 5.6.0
torchao: 0.9.0
numpy: 1.26.4
aiohttp: 3.13.5
fastapi: 0.123.10
huggingface_hub: 1.11.0
interegular: 0.3.3
modelscope: 1.35.3
orjson: 3.11.8
outlines: 0.1.11
packaging: 26.0
psutil: 7.2.2
pydantic: 2.12.5
python-multipart: 0.0.24
pyzmq: 27.1.0
uvicorn: 0.44.0
uvloop: 0.22.1
vllm: Module Not Found
xgrammar: 0.2.0
openai: 2.6.1
tiktoken: 0.12.0
anthropic: 0.89.0
litellm: Module Not Found
torchcodec: Module Not Found
torch_npu: 2.10.0rc2
sgl-kernel-npu: 2026.3.1
deep_ep: 1.0.0+188c7f4b.cann.8.5.0.b232
Ascend Topology:
           Phy-ID0    Phy-ID1    Phy-ID2    Phy-ID3    Phy-ID4    Phy-ID5    Phy-ID6    Phy-ID7    Phy-ID8    Phy-ID9    Phy-ID10   Phy-ID11   Phy-ID12   Phy-ID13   Phy-ID14   Phy-ID15
Phy-ID0    X          SIO        HCCS_SW    HCCS_SW    HCCS_SW    HCCS_SW    HCCS_SW    HCCS_SW    HCCS_SW    HCCS_SW    HCCS_SW    HCCS_SW    HCCS_SW    HCCS_SW    HCCS_SW    HCCS_SW
Phy-ID1    SIO        X          HCCS_SW    HCCS_SW    HCCS_SW    HCCS_SW    HCCS_SW    HCCS_SW    HCCS_SW    HCCS_SW    HCCS_SW    HCCS_SW    HCCS_SW    HCCS_SW    HCCS_SW    HCCS_SW
Phy-ID2    HCCS_SW    HCCS_SW    X          SIO        HCCS_SW    HCCS_SW    HCCS_SW    HCCS_SW    HCCS_SW    HCCS_SW    HCCS_SW    HCCS_SW    HCCS_SW    HCCS_SW    HCCS_SW    HCCS_SW
Phy-ID3    HCCS_SW    HCCS_SW    SIO        X          HCCS_SW    HCCS_SW    HCCS_SW    HCCS_SW    HCCS_SW    HCCS_SW    HCCS_SW    HCCS_SW    HCCS_SW    HCCS_SW    HCCS_SW    HCCS_SW
Phy-ID4    HCCS_SW    HCCS_SW    HCCS_SW    HCCS_SW    X          SIO        HCCS_SW    HCCS_SW    HCCS_SW    HCCS_SW    HCCS_SW    HCCS_SW    HCCS_SW    HCCS_SW    HCCS_SW    HCCS_SW
Phy-ID5    HCCS_SW    HCCS_SW    HCCS_SW    HCCS_SW    SIO        X          HCCS_SW    HCCS_SW    HCCS_SW    HCCS_SW    HCCS_SW    HCCS_SW    HCCS_SW    HCCS_SW    HCCS_SW    HCCS_SW
Phy-ID6    HCCS_SW    HCCS_SW    HCCS_SW    HCCS_SW    HCCS_SW    HCCS_SW    X          SIO        HCCS_SW    HCCS_SW    HCCS_SW    HCCS_SW    HCCS_SW    HCCS_SW    HCCS_SW    HCCS_SW
Phy-ID7    HCCS_SW    HCCS_SW    HCCS_SW    HCCS_SW    HCCS_SW    HCCS_SW    SIO        X          HCCS_SW    HCCS_SW    HCCS_SW    HCCS_SW    HCCS_SW    HCCS_SW    HCCS_SW    HCCS_SW
Phy-ID8    HCCS_SW    HCCS_SW    HCCS_SW    HCCS_SW    HCCS_SW    HCCS_SW    HCCS_SW    HCCS_SW    X          SIO        HCCS_SW    HCCS_SW    HCCS_SW    HCCS_SW    HCCS_SW    HCCS_SW
Phy-ID9    HCCS_SW    HCCS_SW    HCCS_SW    HCCS_SW    HCCS_SW    HCCS_SW    HCCS_SW    HCCS_SW    SIO        X          HCCS_SW    HCCS_SW    HCCS_SW    HCCS_SW    HCCS_SW    HCCS_SW
Phy-ID10   HCCS_SW    HCCS_SW    HCCS_SW    HCCS_SW    HCCS_SW    HCCS_SW    HCCS_SW    HCCS_SW    HCCS_SW    HCCS_SW    X          SIO        HCCS_SW    HCCS_SW    HCCS_SW    HCCS_SW
Phy-ID11   HCCS_SW    HCCS_SW    HCCS_SW    HCCS_SW    HCCS_SW    HCCS_SW    HCCS_SW    HCCS_SW    HCCS_SW    HCCS_SW    SIO        X          HCCS_SW    HCCS_SW    HCCS_SW    HCCS_SW
Phy-ID12   HCCS_SW    HCCS_SW    HCCS_SW    HCCS_SW    HCCS_SW    HCCS_SW    HCCS_SW    HCCS_SW    HCCS_SW    HCCS_SW    HCCS_SW    HCCS_SW    X          SIO        HCCS_SW    HCCS_SW
Phy-ID13   HCCS_SW    HCCS_SW    HCCS_SW    HCCS_SW    HCCS_SW    HCCS_SW    HCCS_SW    HCCS_SW    HCCS_SW    HCCS_SW    HCCS_SW    HCCS_SW    SIO        X          HCCS_SW    HCCS_SW
Phy-ID14   HCCS_SW    HCCS_SW    HCCS_SW    HCCS_SW    HCCS_SW    HCCS_SW    HCCS_SW    HCCS_SW    HCCS_SW    HCCS_SW    HCCS_SW    HCCS_SW    HCCS_SW    HCCS_SW    X          SIO  
Phy-ID15   HCCS_SW    HCCS_SW    HCCS_SW    HCCS_SW    HCCS_SW    HCCS_SW    HCCS_SW    HCCS_SW    HCCS_SW    HCCS_SW    HCCS_SW    HCCS_SW    HCCS_SW    HCCS_SW    SIO        X    

Legend:

  X    = Self
  SYS  = Path traversing PCIe and NUMA nodes. Nodes are connected through SMP, such as QPI, UPI.
  PHB  = Path traversing PCIe and the PCIe host bridge of a CPU.
  PIX  = Path traversing a single PCIe switch
  PXB  = Path traversing multipul PCIe switches
  HCCS = Connection traversing HCCS.
  SIO  = Path traversing the SIO bus
  HCCS_SW = Connection traversing HCCS through a switch
  NA   = Unknown relationship.

ulimit soft: 1024

### PR DESCRIPTION
**Pull request by OrangeRedeng**
## Description

Fix Wan2.2-T2V-A14B-Diffusers-w8a8/w8a8/mxfp8 quant scheme recognition on NPU by threading
`reverse_param_names_mapping` through the ModelSlim config stack.

## Motivation

Fix of https://github.com/sgl-project/sglang/issues/24518. After PR #23625 reshaped quantized‑diffusion prefixes, the Wan2.2‑W8A8 model stopped loading (`No modelslim compatible scheme`). The bug is that `_get_scheme_from_parts` uses the **model’s internal layer name** (e.g. `blocks.0.self_attn.to_q`) to look up the quant description, but the quant config file uses **architecture‑canonical names** (e.g. `blocks.0.attn1.to_q`).

The WanVideo arch‑config already carries a `reverse_param_names_mapping` that translates internal names back to canonical names, but the ModelSlim initialisation path silently discarded this mapping.

## Modifications

- **`modelslim.py`** – accept `reverse_param_names_mapping` in `__init__` and `from_config`, build a mapper via `get_param_names_mapping`, and use it in `_get_scheme_from_parts` before querying `quant_model_description.json`.
- **`transformer_load_utils.py`** – read `reverse_param_names_mapping` from the arch config and pass it into `get_quant_config`.
- **`quantization_utils.py`** – accept and forward `reverse_param_names_mapping`.
- **`fsdp_load.py`** – add missing quant parameter suffixes (`input_offset`, `quant_bias`, `deq_scale`) to the sharding allow‑list; mark sharded tensors with `requires_grad=False` to avoid meta‑tensor copy errors.
- **`npu/utils.py`** – update a performance warning to mention `--dit-cpu-offload false`.

## Accuracy Tests

Command:
```SGLANG_CACHE_DIT_FN=2 SGLANG_CACHE_DIT_BN=1 SGLANG_CACHE_DIT_WARMUP=4 SGLANG_CACHE_DIT_RDT=0.4 SGLANG_CACHE_DIT_MC=4 SGLANG_CACHE_DIT_TAYLORSEER=true SGLANG_CACHE_DIT_TS_ORDER=2 SGLANG_CACHE_DIT_ENABLED=true  sglang generate --model-path ./weights/Wan2.2-T2V-A14B-Diffusers-w8a8/ --prompt "Two anthropomorphic cats in comfy boxing gear and bright gloves fight intensely on a spotlighted stage." --height 720 --width 1280 --tp-size 2 --sp-degree 2 --num-gpus 4 --num-frames 81 --num-inference-steps 40 --warmup true```

https://github.com/user-attachments/assets/678e4823-0c70-47f5-82f5-e1f0e0f52a51

## Speed Tests and Profiling

image

## Checklist

- [x] Format your code according to the [Format code with pre-commit](https://docs.sglang.io/developer_guide/contribution_guide.html#format-code-with-pre-commit).
- [ ] Add unit tests according to the [Run and add unit tests](https://docs.sglang.io/developer_guide/contribution_guide.html#run-and-add-unit-tests).
- [ ] Update documentation according to [Write documentations](https://docs.sglang.io/developer_guide/contribution_guide.html#write-documentations).
- [x] Provide accuracy and speed benchmark results according to [Test the accuracy](https://docs.sglang.io/developer_guide/contribution_guide.html#test-the-accuracy) and [Benchmark the speed](https://docs.sglang.io/developer_guide/contribution_guide.html#benchmark-the-speed).
- [x] Follow the SGLang code style [guidance](https://docs.sglang.io/developer_guide/contribution_guide.html#code-style-guidance).

## Review and Merge Process

1. Ping Merge Oncalls to start the process. See the [PR Merge Process](https://github.com/sgl-project/sglang/blob/main/.github/MAINTAINER.md#pull-request-merge-process).
2. Get approvals from [CODEOWNERS](https://github.com/sgl-project/sglang/blob/main/.github/CODEOWNERS) and other reviewers.
3. Trigger CI tests with [comments](https://docs.sglang.io/developer_guide/contribution_guide.html#how-to-trigger-ci-tests) or contact authorized users to do so.
   - Common commands include `/tag-and-rerun-ci`, `/tag-run-ci-label`, `/rerun-failed-ci`
4. After green CI and required approvals, ask Merge Oncalls or people with Write permission to merge the PR.

### CODE REVIEW

File: python/sglang/multimodal_gen/runtime/layers/quantization/modelslim.py [hunk 1]
```diff
@@ -21,11 +21,12 @@
 )
 
 if TYPE_CHECKING:
-    from sglang.srt.layers.quantization.modelslim.modelslim import ModelSlimConfig
     from sglang.srt.layers.quantization.modelslim.schemes import (
         ModelSlimLinearScheme,
     )
 
+from sglang.multimodal_gen.runtime.loader.utils import get_param_names_mapping
```
Comment (ping1jing2) [inline python/sglang/multimodal_gen/runtime/layers/quantization/modelslim.py:28]: maybe we should move `import` above `if`

**Review by ping1jing2 — APPROVED** [review]

File: python/sglang/multimodal_gen/runtime/layers/quantization/modelslim.py (hunk 1 above)
Comment (OrangeRedeng) [reply python/sglang/multimodal_gen/runtime/layers/quantization/modelslim.py:28]: Tried, not worked

**Comment by ping1jing2** [comment]
this is a Diffusion related PR and all GPU CIs passed, so i merged it
